src/app/search/search.component.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { Subject } from 'rxjs';
+import { ActivatedRoute, Router } from '@angular/router';
+import { cit, create, cms, provide } from '../../testing';
+import { RouterStub, ActivatedRouteStub } from '../../testing/stub.router';
+import { SearchComponent } from './search.component';
+
+import { AuthService } from '../core/auth/auth.service';
+import { CmsService } from '../core/cms/cms.service';
+
+let authToken = new Subject<string>();
+let cmsToken: string = null;
+let activatedRoute: ActivatedRouteStub = new ActivatedRouteStub();
+
+describe('SearchComponent', () => {
+
+  create(SearchComponent);
+
+  provide(AuthService, {token: authToken, setToken: (t) => cmsToken = t});
+  provide(CmsService, cms);
+  provide(Router, RouterStub);
+  provide(ActivatedRoute, activatedRoute);
+
+  let auth;
+  beforeEach(() => {
+    activatedRoute.testParams = {};
+    auth = cms.mock('prx:authorization', {});
```

Comment:
I did add the router.stub upon getting the 'No provider for Router!' error, but yeah looks like I can remove AuthService and CmsService